full-stack-practice/client/src/lib/read.ts

```diff
@@ -0,0 +1,9 @@
+import { products, type Product } from './data';
```

Comment:
You aren't using this file, so you could delete it.

However, my suggestion is that you instead move all the `fetch` calls in Catalog and Details into the functions below. That will make your code cleaner and easier to work with.